gamemodes/terrortown/entities/weapons/weapon_tttbasegrenade.lua

```diff
@@ -84,6 +85,107 @@ function SWEP:SecondaryAttack()
 
 end
 
+---
+-- @param Vector P The initial position for this iteration cycle.
+-- @param number V The launch velocity for this iteration cycle.
+-- @param number G The gravity multiplier for this iteration cycle.
+-- @param number T The step size for the current iteration cycle.
+-- @return Vector The next location to iterate from.
+local function PositionFromPhysicsParams(P, V, G, T)
+	local A = G * physenv.GetGravity()
+
+	return P + (V * T + 0.5 * A * T ^ 2)
+end
+
+if CLIENT then
+	---
+	-- @realm client
+	local trajectory_ui = CreateConVar("ttt2_grenade_trajectory_ui", 0, FCVAR_ARCHIVE)
+
+	local function AlphaLerp(from, frac, max)
+		local fr = frac ^ 0.5
+		return ColorAlpha(from, Lerp(fr, 0, math.min(max, 255)))
+	end
+
+	---
+	-- @param Player ply
+	-- @realm client
+	function SWEP:DrawDefaultThrowPath(ply)
+		local owner = self:GetOwner()
+		local pos, _ = self:GetViewModelPosition(owner:EyePos(), owner:EyeAngles())
+
+		local src, thr = self:GetThrowVelocity()
+		local P = pos - ply:EyePos() + src
+		local V = thr
+		local G = 1
+		local step = 0.005
+		local lastpos = PositionFromPhysicsParams(P, V, G, step)
+		local frac = (SysTime() % 1) * 2
+		local i = frac > 1 and 1 or 0
+		frac = frac - math.floor(frac)
+		local arc_color = appearance.ShouldUseGlobalFocusColor() and appearance.GetFocusColor() or self:GetOwner():GetRoleColor()
+		local alpha = math.Round(GetConVar("ttt_crosshair_opacity"):GetFloat() * 255)
```

Comment:
could you please give the variables clearer names? I know that in base TTT2 there are also quite a few badly named variables, but I have no idea what P, V, G and i are.

Stuff like thr should probably be thrower? or is it something different than the thrower entity?